pkg/action/process.go

```diff
@@ -0,0 +1,101 @@
+package action
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+type Process struct {
+	PID  int
+	Path string
+}
+
+// GetAllProcessPaths is an exported function that retrieves the process commands.
+func GetAllProcessPaths() ([]Process, error) {
+	switch runtime.GOOS {
+	case "linux", "darwin":
+		return getUnixProcessPaths()
+	case "windows":
+		return getWindowsProcessPaths()
+	default:
+		return nil, fmt.Errorf("unsupported operating system")
+	}
+}
+
+// getUnixProcessPaths is a UNIX-focused function to retrieve PIDs and their respective commands.
+func getUnixProcessPaths() ([]Process, error) {
+	cmd := exec.Command("ps", "-e", "-o", "pid=,comm=")
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	uniquePaths := make(map[string]Process)
+	lines := strings.Split(string(output), "\n")
+	// Iterate through the output lines
+	for _, line := range lines {
+		fields := strings.Fields(line)
+		if len(fields) >= 2 {
+			pid, err := strconv.Atoi(fields[0])
+			if err != nil {
+				continue
+			}
+			path := strings.TrimSpace(strings.Join(fields[1:], " "))
+			if _, exists := uniquePaths[path]; !exists && isValidPath(path) {
+				uniquePaths[path] = Process{PID: pid, Path: path}
+			}
+		}
+	}
+
+	return mapToSlice(uniquePaths), nil
+}
+
+// getWindowsProcessPaths is a Windows-focused function to retrieve PIDs and their respective commands.
+func getWindowsProcessPaths() ([]Process, error) {
+	cmd := exec.Command("wmic", "process", "get", "ProcessId,ExecutablePath")
```

Comment:
For reference: https://superuser.com/a/1747592

I can potentially test this out later.